browser-sign-in/OktaBrowserSignIn/AppDelegate.swift

```diff
@@ -15,28 +15,78 @@
  */
 
 import UIKit
-import OktaOidc
+import WebAuthenticationUI
 
 @UIApplicationMain
 class AppDelegate: UIResponder, UIApplicationDelegate {
     
     var window: UIWindow?
-
+    var configForUITests: [String: String]? {
+        let env = ProcessInfo.processInfo.environment
+        guard let oktaURL = env["OKTA_URL"], oktaURL.count > 0,
+              let clientID = env["CLIENT_ID"],
+              let redirectURI = env["REDIRECT_URI"],
+              let logoutRedirectURI = env["LOGOUT_REDIRECT_URI"] else {
+            return nil
+        }
+        return ["issuer": "\(oktaURL)/oauth2/default",
+                "clientId": clientID,
+                "redirectUri": redirectURI,
+                "logoutRedirectUri": logoutRedirectURI,
+                "scopes": "openid profile offline_access"
+        ]
+    }
+    
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        self.window = UIWindow(frame: UIScreen.main.bounds)
+        
+        if ProcessInfo.processInfo.arguments.contains("--reset-keychain") {
+            try? Keychain.Search().delete()
+        }
+        window?.rootViewController = self.getRootViewController()
+        window?.makeKeyAndVisible()
         return true
     }
     
-    func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any]) -> Bool {
-        return false
+    func getRootViewController() -> UIViewController? {
+        let storyboard = UIStoryboard(name: "Main", bundle: nil)
+        guard let _ = Credential.default else {
+            let welcomeViewController = storyboard.instantiateViewController(withIdentifier: "SignIn") as? WelcomeViewController
+            //Setup for UI Tests
+            if let configForUITests = configForUITests {
+                welcomeViewController?.auth = WebAuthentication(
```

Comment:
You shouldn't need to assign this to the view controller's property. It's not clear in [the documentation for `shared`](https://okta.github.io/okta-mobile-swift/development/webauthenticationui/documentation/webauthenticationui/webauthentication/shared) yet (I've taken a note to update the docs), but if you construct a WebAuthentication instance programmatically, it will become the shared instance. The idea being that the Okta.plist is an optional convenience, but `WebAuthentication.shared` can always be used by a developer, regardless of how they choose to configure their settings.

So you can do the following:

```
let _ = WebAuthentication(issuer: ..., etc...)
```

This means you don't need the `else` condition, because WelcomeViewController internally will still just use `WebAuthentication.shared`

Reply:
Yes thanks for clarifying this. I thought about this for a bit since the shared instance had a private "setter" It was abit confusing but it makes sense now.

Reply:
The usage seemed more like but thanks for clarifying
- If the okta.plist is present -> you can use the `WebAuthentication.shared`
- else you have to programmatically create an instance and manage it.